docs/data-masking-function-list.md

```diff
@@ -0,0 +1,888 @@
+A random PAN string.
+
+### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT gen_rnd_pan();
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +-------------------+
+    | gen_rnd_pan()     |
+    +-------------------+
+    | DAJPC4150PHYTYB3  |
+    +-------------------+
+    ```
+
+## gen_rnd_ssn()
+
+
+
+Generates a United States Social Security Account Number (SSN).
+
+
+### Parameters
+
+None
+
+
+### Returns
+
+A SSN string in a `***-**-****` format. The first three numbers are greater than 900, which are outside the range for legitimate numbers.
+
+### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT gen_rnd_ssn();
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +----------------+
+    | gen_rnd_ssn()  |
+    +----------------+
+    | 970-03-0370    |
+    -----------------+
+    ```
+
+## gen_rnd_uk_nin()
+
+
+
+Generates a United Kingdom National Insurance Number (NIN).
+
+!!! important
+
+    Generating the NIN should only be used for testing. The function does not check if the generated value is a legitimate primary account number. If you must publish the result, consider using [`mask_uk_nin`](#mask_uk_ninstr-mask_char) to mix up the result.
+
+### Parameters
+
+None.
+
+### Returns
+
+A NIN string.
+
+### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT gen_rnd_uk_nin();
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +----------------------+
+    | gen_rnd_uk_nin()     |
+    +----------------------+
+    | BK189806             |
+    +----------------------+
+    ```
+
+
+## gen_rnd_us_phone()
+
+
+
+Generates a United States phone number with the `555` area code.
+
+### Parameters
+
+None
+
+### Returns
+
+Returns a United States phone number.
+
+### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT gen_rnd_us_phone();
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +--------------------+
+    | gen_rnd_us_phone() |
+    +--------------------+
+    | 1-555-249-2029     |
+    +--------------------+
+    ```
+
+## gen_rnd_uuid()
+
+
+
+Generates a Universally Unique Identifier (UUID).
+
+### Parameters
+
+None.
+
+### Returns
+
+Returns a UUID as a string.
+
+### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT gen_rnd_uuid();
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +------------------------------------+
+    | gen_rnd_uuid()                     |
+    +------------------------------------+
+    |9a3b642c-06c6-11ee-be56-0242ac120002|
+    +------------------------------------+
+    ```
+
+## mask_canada_sin(str [,mask_char])
+
+
+
+Masks a Canada Social Insurance Number (SIN).
+
+### Parameters
+
+| Parameter | Optional | Description | Type |
+| --- | --- | --- | --- |
+| str | No | The string to be masked | String |
+| mask_char | Yes | The masking character | String |
+
+The `str` accepts an alphanumeric string.
+
+If the `mask_char` is not specified, the default is `X`.
+
```

Comment:
`mask_char` can be a multibyte character in any character set (not necessarily identical to the character set of `str`)